src/main/java/christmas/domain/discount/service/DiscountService.java

```diff
@@ -0,0 +1,43 @@
+package christmas.domain.discount.service;
+
+
+import christmas.domain.discount.domain.DiscountDetails;
+import christmas.domain.discount.policy.ChristmasDiscount;
+import christmas.domain.discount.policy.Discount;
+import christmas.domain.discount.policy.GiftDiscount;
+import christmas.domain.discount.policy.StarDayDiscount;
+import christmas.domain.discount.policy.WeekDayDiscount;
+import christmas.domain.discount.policy.WeekendDiscount;
+import christmas.domain.order.domain.Order;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DiscountService {
+    
+    private final List<Discount> discounts;
+
+    public DiscountService(ChristmasDiscount christmasDiscount,
+                           WeekDayDiscount weekDayDiscount,
+                           WeekendDiscount weekendDiscount,
+                           StarDayDiscount starDayDiscount,
+                           GiftDiscount giftDiscount) {
+        this.discounts = List.of(christmasDiscount,
+                weekDayDiscount,
+                weekendDiscount,
+                starDayDiscount,
+                giftDiscount);
+    }
```

Comment:
```suggestion
    private final List<Discount> discounts;

    public DiscountService(Discount... discounts) {
        this.discounts = List.of(discounts);
    }
```

생성자에 구현체가 순서대로 들어가있고 구현체에 따라 변경이 많은 점을 생각해보면, 인터페이스로 멋지게 추상화를 해놓았지만 그 효과를 못 활용하는 느낌입니다.
저렇게 어떤 구현체가 들어올 수 있도록 작성하는 것은 어떠신가요?

Reply:
그리고 네이밍을 discountPolicy 로 하는 것이 좀 더 읽기 편할 것 같은데 어떻게 생각하시나요

Reply:
222222 가변인자에 대해서 학습해보시기를 추천드립니다

Reply:
제가 Controller, Service, Repository 이런 레이어에 너무 집착하여 객체지향적인 측면을 많이 놓친 것 같습니다. 할인 정책들이 모여있으니 할인정책들을 포괄하는 역할의 타입으로 DiscountPolicy가 적절할 것 같습니다. 감사합니다!